### ISSUE TITLE
A question about sem_feat

### ISSUE THREAD

**Issue by shenyehui** [IRMVLab/SNI-SLAM#6]
Dear author, thank you very much for your work! Regarding the processing of sem_feat and gt_label after looking at the code I have a question, in Figure2 the fs is obtained from Feature Extractior and the gt_label used to do Semantic_loss is obtained from the whole Segmentation network, but looking at the code both are obtained from the Segmentation network?

**Comment by shenyehui** [comment]
#if self.mode == 'mapping':
            for i in range(3):
                out = self.segmentation_conv[i](out)
      Why is this place 3 and not 4? if it's 3, the last layer
      nn.Conv2d(dim, self.num_class, (3, 3), padding=(1, 1)) would not be involved in the calculation the output would be (B,dim.H,W) not (B, num_class, H,W) is it set up this way on purpose?

**Comment by zstsandy** [comment]
In semantic optimization, we have two types of losses: one is the cross-entropy loss, which is constructed using semantic segmentation results(labels) and the rendering results; the other one is the feature loss, which is constructed using rendering features and the extracted features.
